scripts/dataset_processing/get_librispeech_data.py

```diff
@@ -93,27 +129,13 @@ def __process_data(data_folder: str, dst_folder: str, manifest_file: str):
 
     for root, dirnames, filenames in os.walk(data_folder):
         for filename in fnmatch.filter(filenames, '*.trans.txt'):
-            files.append((os.path.join(root, filename), root))
-
-    for transcripts_file, root in tqdm(files):
-        with open(transcripts_file, encoding="utf-8") as fin:
-            for line in fin:
-                id, text = line[: line.index(" ")], line[line.index(" ") + 1 :]
-                transcript_text = text.lower().strip()
-
-                # Convert FLAC file to WAV
-                flac_file = os.path.join(root, id + ".flac")
-                wav_file = os.path.join(dst_folder, id + ".wav")
-                if not os.path.exists(wav_file):
-                    Transformer().build(flac_file, wav_file)
-                # check duration
-                duration = subprocess.check_output("soxi -D {0}".format(wav_file), shell=True)
-
-                entry = {}
-                entry['audio_filepath'] = os.path.abspath(wav_file)
-                entry['duration'] = float(duration)
-                entry['text'] = transcript_text
-                entries.append(entry)
+            files.append(os.path.join(root, filename))
+
+    with multiprocessing.Pool(num_workers) as p:
+        processing_func = functools.partial(__process_transcript, dst_folder=dst_folder)
+        results = p.imap_unordered(processing_func, files)
```

Comment:
This is unordered map - changes order of validation data files which will make it hard to process data (equivalent to shuffling validation and text data). Use joblib.Parallel or mp.map() for ordered results.

Reply:
Thanks for the catch about the order.

I decided to got with imap. It guarantees order and I checked on my machine that the order is preserved. 
Compared to joblib, it is much easier to couple tqdm with it. And having progress bar is very informative for these long scripts.